app/src/server/api/routers/datasets.router.ts

```diff
@@ -53,38 +55,39 @@ export const datasetsRouter = createTRPCRouter({
   getTrainingCosts: protectedProcedure
     .input(
       z.object({
-        id: z.string(),
+        datasetId: z.string(),
+        baseModel: baseModel,
         filters: filtersSchema,
-        selectedPruningRuleIds: z.array(z.string()),
+        pruningRuleIds: z.array(z.string()),
       }),
     )
     .query(async ({ input, ctx }) => {
-      const { id, filters, selectedPruningRuleIds } = input;
+      const { datasetId, filters, baseModel, pruningRuleIds } = input;
 
       const { projectId } = await prisma.dataset.findUniqueOrThrow({
-        where: { id },
+        where: { id: datasetId },
       });
 
       await requireCanViewProject(projectId, ctx);
 
-      const trainingEntryStats = await constructDatasetEntryFiltersQuery({ filters, datasetId: id })
+      // Ensure pruningRuleIds always contains at least one element, using a value that won't match anything
+      const safePruningRuleIds = pruningRuleIds.length > 0 ? pruningRuleIds : [uuidv4()];
+
+      const trainingEntryStats = await constructDatasetEntryFiltersQuery({ filters, datasetId })
         .where("de.split", "=", "TRAIN")
         .where("de.output", "is not", null)
-        .select((eb) => [
-          sql<number>`count(*)::int`.as("numEntries"),
+        .leftJoin("PruningRuleMatch as prm", "prm.datasetEntryId", "de.id")
+        .leftJoin("PruningRule as pr", "prm.pruningRuleId", "pr.id")
+        .where("pr.id", "in", safePruningRuleIds)
```

Comment:
You can actually make this where clause part of the join statement, and postgresql will be able to plan the query more efficiently.

https://kysely.dev/docs/examples/JOIN/complex-join